### ISSUE TITLE
Rule Change: Add `allowFunctions` option to "use-baseline" rule

### ISSUE THREAD

**Issue by AlexandreGaubil** [eslint/css#292]
### What rule do you want to change?

use-baseline

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```css
input {
    background-color: light-dark(#fff, #000);
}
```

### What does the rule currently do for this code?

> Type 'light-dark' is not a 2023 available baseline feature

### What will the rule do after it's changed?

It would not report an error.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

By adding a `allowFunctions` to the `options` of "use-baseline", we would be able to allow certain functions (such as `light-dark`), similarly to how we can already allow some properties (`allowProperties`).

This was mentioned in https://github.com/eslint/css/issues/210#issuecomment-3148167404 but was not implemented.

Our project uses a lot of these functions. As we use PostCSS, it is transformed into valid baseline CSS. This is preventing us from enabling this rule, as it causes too many warnings in our codebase.
